modules/apps/segments/segments-web/src/main/java/com/liferay/segments/web/internal/display/context/SegmentsExperienceSelectorDisplayContext.java

```diff
@@ -71,48 +73,12 @@ public SegmentsExperienceSelectorDisplayContext(
 			WebKeys.THEME_DISPLAY);
 	}
 
-	public JSONObject getSegmentsExperienceSelectedJSONObject()
-		throws PortalException {
-
-		JSONObject segmentsExperienceSelectedJSONObject =
-			_jsonFactory.createJSONObject();
-
-		SegmentsExperience segmentsExperience =
-			_fetchSegmentsExperienceFromRequest();
-
-		if (segmentsExperience != null) {
-			segmentsExperienceSelectedJSONObject =
-				_getSegmentsExperienceJSONObject(
-					segmentsExperience.getSegmentsExperienceId());
-
-			segmentsExperienceSelectedJSONObject.put(
-				"segmentsExperienceName",
-				_getSelectedSegmentsExperienceName(segmentsExperience));
-		}
-
-		return segmentsExperienceSelectedJSONObject;
-	}
-
-	public JSONArray getSegmentsExperiencesJSONArray() throws PortalException {
-		if (_segmentsExperiencesJSONArray != null) {
-			return _segmentsExperiencesJSONArray;
-		}
-
-		JSONArray segmentsExperiencesJSONArray = _jsonFactory.createJSONArray();
-
-		List<SegmentsExperience> segmentsExperiences =
-			_segmentsExperienceLocalService.getSegmentsExperiences(
-				_themeDisplay.getScopeGroupId(),
-				_portal.getClassNameId(Layout.class.getName()),
-				_themeDisplay.getPlid(), true);
-
-		for (SegmentsExperience segmentsExperience : segmentsExperiences) {
-			segmentsExperiencesJSONArray.put(
-				_getSegmentsExperienceJSONObject(
-					segmentsExperience, segmentsExperiences));
-		}
-
-		return segmentsExperiencesJSONArray;
+	public Map<String, Object> getData() throws PortalException {
+		return HashMapBuilder.<String, Object>put(
+			"experiences", _getSegmentsExperiencesJSONArray()
+		).put(
+			"selectedExperience", _getSegmentsExperienceSelectedJSONObject()
```

Comment:
the JSON should be selectedSegmentsExperiences

Reply:
@cgoncas , shouldn't it be in singular? `selectedSegmentsExperience` 🤔

Reply:
Yes it is :)

Reply:
Changed!